elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchRel.java

```diff
@@ -54,6 +55,12 @@
      */
     final List<Map.Entry<String, RelFieldCollation.Direction>> sort = new ArrayList<>();
 
+    /**
+     * Sorting missing values.
+     * @see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html#_missing_values">Missing Values</a>
+     */
+    final Map<String, RelFieldCollation.NullDirection> nullsSort = new HashMap<>();
```

Comment:
I suspect this use Map have some hidden problems when handle have duplicate columns. I suggest we use  `List<Map.Entry<String, RelFieldCollation.Direction>>` to handle this.

Reply:
While sorting duplicated columns, the latter ones will be treated as redundant.
Since sort and nulls sort are corresponding for one field, I've changed the Collection to store NULLS SORT DIRECTION. Thanks @NobiGo .